include/circt/Dialect/HW/HWAttributesNaming.td

```diff
@@ -17,32 +17,65 @@ include "circt/Dialect/HW/HWDialect.td"
 include "mlir/IR/AttrTypeBase.td"
 
 def InnerRefAttr : AttrDef<HWDialect, "InnerRef"> {
-  let summary = "Refer to a name inside a module";
+  let summary = "Refer to a name inside a list of scopes";
   let description = [{
-    This works like a symbol reference, but to a name inside a module.
+    This works like a symbol reference, but to a name inside a list of `InnerSymbolTable`'s.
   }];
   let mnemonic = "innerNameRef";
-  let parameters = (ins "::mlir::FlatSymbolRefAttr":$moduleRef,
-                        "::mlir::StringAttr":$name);
+  let parameters = (ins ArrayRefParameter<"::mlir::StringAttr">:$path);
 
   let builders = [
-    AttrBuilderWithInferredContext<(ins "::mlir::StringAttr":$module,
-                                        "::mlir::StringAttr":$name), [{
-      return $_get(
-          module.getContext(), mlir::FlatSymbolRefAttr::get(module), name);
-    }]>
+
+    AttrBuilderWithInferredContext<(ins "::mlir::ArrayRef<::mlir::StringAttr>":$pathAttr), [{
+      assert(!pathAttr.empty() && "Empty path");
+      assert(pathAttr.front() && "Null path element");
```

Comment:
nit: Probably should check something like:
`llvm::all_of(pathAttr, [](auto& e) { assert(e); });`
any null element is malformed.